### PR TITLE
T11/admin insert data view

### PR DESCRIPTION
**Pull request by promatty**
The admin panel, which allows AdminUser members to choose Users from the database, and add Sessions to them.

### CODE REVIEW

File: src/components/admin/AdminForm.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+export default function AdminForm() {
+  const [users, setUsers] = useState<User[]>([]);
+  const [searchQuery, setSearchQuery] = useState<string>("");
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [formSubmitted, setFormSubmitted] = useState<boolean>(false);
+  const { register, handleSubmit, reset } = useForm<SessionFormData>();
+
+  //to get all the users in our database
+  useEffect(() => {
```
Comment (burtonjong) [inline src/components/admin/AdminForm.tsx:28]: use tanstack

Comment (burtonjong) [inline src/components/admin/AdminForm.tsx]: use tanstack

File: src/components/admin/AdminForm.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+          ))}
+        </ul>
+      </div>
+
+      {selectedUser && (
+        <div style={{ flex: "1" }}>
+          <h3>
+            Add a Session for {selectedUser.firstName} {selectedUser.lastName}
+          </h3>
+          <form
```
Comment (burtonjong) [inline src/components/admin/AdminForm.tsx:121]: please just use tailwind

File: src/components/admin/AdminForm.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+              display: "flex",
+              flexDirection: "column",
+              gap: "10px",
+              padding: "20px",
+              border: "1px solid #ccc",
+              borderRadius: "4px",
+              backgroundColor: "#fafafa",
+            }}
+          >
+            <div style={{ marginBottom: "10px" }}>
```
Comment (burtonjong) [inline src/components/admin/AdminForm.tsx]: tailwind

File: src/components/admin/AdminForm.tsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
+"use client";
+
+import { generateClient } from "aws-amplify/data";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
```
Comment (burtonjong) [inline src/components/admin/AdminForm.tsx]: fix eslint error here

File: src/components/Header.tsx [hunk 5]
```diff
@@ -49,15 +53,26 @@ export default function Header({
...
-  const routes = defaultRoutes.filter(
+  const loggedInRoutes = defaultRoutes.filter(
     (route) => route.loggedIn === authCheck || route.loggedIn === undefined,
   );
 
+  let renderedRoutes = loggedInRoutes;
+
+  if (user.role !== "AdminUser") {
+    renderedRoutes = loggedInRoutes.filter(
+      (route) => route.name != "Admin Panel",
```
Comment (burtonjong) [inline src/components/Header.tsx]: fix eslint

**Review by burtonjong — CHANGES_REQUESTED** [review]

**Comment by promatty** [comment]
hit me with a "lgtm" for a cookie :)

**Review by burtonjong — APPROVED** [review]

**Comment by burtonjong** [comment]
lgtm just had to deal with merge conflicts as long as eslint passes

**Comment by promatty** [comment]
YIPPEE!!!!

**Pull request merged**